### DISCUSSION TITLE
[IntelliSense] Unwanted suggestions in JSX components

### DISCUSSION THREAD

**Discussion by OnkelTem** [tailwindlabs/tailwindcss#9532] [Help]
**What version of Tailwind CSS IntelliSense are you using?**

0.8.7

**What version of Tailwind CSS are you using?**

3.1.8

**What package manager are you using?**

npm

**What operating system are you using?**

Kubuntu 22.04

**Reproduction URL**

—

**Describe your issue**

The problem is that whenever I try to list the properties of a JSX component by pressing `Ctrl-Space`, I get the output spoiled with the CSS suggestions instead:

![image](https://user-images.githubusercontent.com/114060/195399697-e25a9a62-ddaa-4c7f-bfe3-9a7f008cbdb3.png)

The only way to get those properties back I found so far is to DISABLE the plugin. Then the base functionality starts working again:

![image](https://user-images.githubusercontent.com/114060/195400221-4cd3b642-0823-4c3c-9700-45d5a524e8f3.png)

I can export my current configuration of the plugin if you tell me how to do it. But I haven't changed anything there AFAIR.

**Comment by bradlc** [comment]
Hey @OnkelTem. This is likely a misconfigured `tailwindCSS.experimental.classRegex` setting. Do you have this configured and if so please can you share it?

**Reply by OnkelTem** [reply]
I really found a rogue piece in my user config :) Not sure where it came from:

```json5

  "tailwindCSS.experimental.classRegex": [
    "tw`([^`]*)", // tw`...`
    "tw=\"([^']*)", // <div tw="..." />
    "tw={\"([^'}]*)", // <div tw={"..."} />
    "tw\\.\\w+`([^`]*)", // tw.xxx`...`
    "tw\\(.*?\\)`([^`]*)" // tw(Component)`...`
  ],
  "tailwindCSS.includeLanguages": {
    "typescript": "javascript",
    "typescriptreact": "javascript"
  },
  "tailwindCSS.classAttributes": [
    "tw"
  ],
```

**Reply by OnkelTem** [reply]
After I commented out everything here, the initial problem was solved, but the plugin stopped working on `tw` attributes, so I probably copied this piece from the [twin.macro](https://github.com/ben-rogerson/twin.macro) docs.

**Reply by OnkelTem** [reply]
Yeah, from here: https://github.com/ben-rogerson/twin.macro/discussions/227

**Reply by OnkelTem** [reply]
I copied that block one more time and it seems to be working now.

Sorry for an invalid report, @bradlc . And many many thanks for the quick reply, I've been struggling with this for two months.

**Comment by bradlc** [comment]
Converted this to a discussion until an issue (if there is one) can be identified. As I said this is likely a rogue `classRegex` configuration. You can see your settings in JSON format by running the `Preferences: Open User Settings (JSON)` command from the VS Code command palette.
